### PR TITLE
AsyncClient refactoring and android-free user tests.

### PR DESCRIPTION
**Pull request by makingthematrix**
There are now wrappers on Koushi's AsyncHttpClient, AsyncHttpRequest and AsyncHttpResponse, and worker classes taking care of processing the request and the response, so they can be mocked in tests. The Request and Response case classes can be converted to respective AsyncHttp* classes and so they are used directly in AsyncClient.

### CODE REVIEW

**Comment by makingthematrix** [comment]
build again

File: zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala [hunk 1]
```diff
@@ -21,101 +21,146 @@ import java.security.cert.X509Certificate
...
+    new Thread(new Runnable {
+      override def run(): Unit = p.complete(Try{ f.get })
+    }).start
+
+    new CancellableFuture(p){
+      override def cancel()(implicit tag: LogTag): Boolean = f.cancel(true)
+    }
+  }
+
+  def toKoushi(wrapper: ClientWrapper): AsyncHttpClient = wrapper match {
```
Comment (wire-android) [inline zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala]: Not fan of this naming. Would rather prefer to use project name, not the author's. 
And even if you think his name is better then we should spell it correctly.

File: zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala [hunk 2]
```diff
@@ -21,101 +21,146 @@ import java.security.cert.X509Certificate
...
-object ClientWrapper extends ClientWrapper {
+object ClientWrapper {
   import Threading.Implicits.Background
 
+  import scala.language.implicitConversions
+
+  implicit def koushiToCancellable[T](f: KFuture[T]): CancellableFuture[T] = {
+    val p = Promise[T]()
+
+    new Thread(new Runnable {
```
Comment (wire-android) [inline zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala]: Please don't create new threads, if you really need blocking operation than use IO background queue (or some special executor).
 
AndroidAsync futures also support callbacks (`KFuture.setCallback`), so there is no need to use blocking `get` here.

File: tests/integration/src/test/scala/com/waz/znet/AsyncClientWebSpec.scala [hunk 3]
```diff
@@ -109,7 +111,7 @@ class AsyncClientWebSpec extends FeatureSpecLike with Matchers with BeforeAndAft
     scenario("post data") {
       val data = (0 to 100).map("test_" + _).mkString(", ").getBytes("utf8")
 
-      Await.result(client(URI.parse("http://posttestserver.com/post.php"), "POST", new BinaryRequestContent(data, "text/plain"), timeout = AsyncClient.DefaultTimeout), 45.seconds) match {
+      Await.result(client(URI.parse("http://posttestserver.com/post.php"), Request.Post("/post.php", new BinaryRequestContent(data, "text/plain"))), 45.seconds) match {
```
Comment (makingthematrix) [inline tests/integration/src/test/scala/com/waz/znet/AsyncClientWebSpec.scala]: OK. I made changes to `Request`. Now the uris are stored inside it as baseUri and resourcePath.

File: zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala (hunk 2 above)
Comment (makingthematrix) [inline zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala]: I think we should talk about it.

File: zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala (hunk 1 above)
Comment (makingthematrix) [inline zmessaging/src/main/scala/com/waz/znet/ClientWrapper.scala]: OK, changed.

**Pull request closed without merging**